lnd.go

```diff
@@ -1275,10 +1279,35 @@ func waitForWalletPassword(cfg *Config, restEndpoints []net.Addr,
 
 		// With the seed, we can now use the wallet loader to create
 		// the wallet, then pass it back to avoid unlocking it again.
-		birthday := cipherSeed.BirthdayTime()
-		newWallet, err := loader.CreateNewWallet(
-			password, password, cipherSeed.Entropy[:], birthday,
+		var (
+			birthday  time.Time
+			newWallet *wallet.Wallet
+			err       error
 		)
+
+		// The unlocker service made sure either the cipher seed or the
+		// extended key is set.
+		if cipherSeed != nil {
+			birthday = cipherSeed.BirthdayTime()
+			newWallet, err = loader.CreateNewWallet(
+				password, password, cipherSeed.Entropy[:],
+				birthday,
+			)
+		} else {
+			// When importing a wallet from its extended private key
+			// we don't know the birthday as that information is not
```

Comment:
Decided against an optional birthday param for extended key (as discussed in the PR body)? I think it could be useful, eg in the case of an existing on-chain wallet, they probably know the date when the wallet was created so could have a birthday hint?

Reply:
Hmm, you're right, it might be useful to have the birthday field. Added it.